.github/workflows/reusable-workflow.yml

```
@@ -89,13 +100,25 @@ jobs:
           CYPRESS_BIDDER_MNEMONIC_INPUT: ${{ inputs.bidder_mnemonic }}
           CYPRESS_BIDDER_ADDRESS_INPUT: ${{ inputs.bidder_address }}
 
-      - name: Archive e2e artifacts
-        uses: actions/upload-artifact@v3
-        if: always()
+      - name: Upload e2e artifacts as workflow artifacts
+        uses: actions/upload-artifact@v4
+        if: ${{ !inputs.is_emerynet_test }}
```

we should be adding `!cancelled()` here as well. this will allow this step to run even if the previous step (the running of test) fails.
```suggestion
        if: ${{ !inputs.is_emerynet_test && !cancelled() }}
```